docs/source/using-discovery/bash.rst

```diff
@@ -0,0 +1,867 @@
+  This is line 3
+
+  sed 's/line 1/Line 1/' file1.txt
+  This is Line 1
+  This is line 2
+  This is line 3
+
+Reference: `sed(1) manual page`_
+
+``gzip`` - Compress or decompress files.
+
+.. code-block:: shell
+  :emphasize-lines: 3
+
+  gzip file1.txt
+  ls
+  file1.txt.gz
+
+.. code-block:: shell
+  :emphasize-lines: 3
+
+  gunzip file1.txt.gz
+  ls
+  file1.txt
+
+Reference: `gzip(1) manual page`_
+
+``tar`` - Create or extract compressed archive files.
+
+.. code-block:: shell
+  :emphasize-lines: 3
+
+  tar cvf archive.tar file1.txt file2.txt
+  ls
+  archive.tar file1.txt file2.txt
+
+.. code-block:: shell
+  :emphasize-lines: 3
+
+  tar xvf archive.tar
+  ls
+  file1.txt file2.txt
+
+Reference: `tar(1) manual page`_
+
+Regular expressions:
+
+.. code-block:: shell
+  :emphasize-lines: 2-3
+
+  grep -E '^[A-Z][a-z]+$' file1.txt
+  John
+  Jane
+
+Parameter expansion:
+
+.. code-block:: shell
+  :emphasize-lines: 3
+
+  name="John Doe"
+  echo ${name// /_}
+  John_Doe
+
+Command line options:
+
+.. code-block:: shell
+  :emphasize-lines: 2-5
+
+  ls -lh
+  total 8.0K
+  drwxrwxr-x 2 user user 4.0K Feb 14 13:29 directory1
+  -rw-rw-r-- 1 user user   12 Feb 14 13:29 file1.txt
+  -rw-rw-r-- 1 user user   14 Feb 14 13:29 file2.txt
+
+Parameter substitution:
+
+.. code-block:: shell
+  :emphasize-lines: 2
+
+  echo ${name:4:3}
+  Doe
+
+Arithmetic operations:
+
+.. code-block:: shell
+  :emphasize-lines: 2
+
+  echo $((2 + 2))
+  4
+
+File tests:
+
+.. code-block:: shell
+  :emphasize-lines: 5
+
+  file=file1.txt
+  if [ -f $file ]; then
+  >   echo "$file is a regular file"
+  > fi
+  file1.txt is a regular file
+
+String tests:
+
+.. code-block:: shell
+  :emphasize-lines: 5
+
+   string="hello"
+   if [ "$string" == "hello" ]; then
+   >   echo "The strings match"
+   > fi
+   The strings match
+
+Command substitution with process substitution::
+
+   diff <(ls /path/to/dir1) <(ls /path/to/dir2)
+
+The next few subsections provide more details on a few advanced bash tools that often come in handy.
+
+rsync
+-----
+The ``rsync`` command is a powerful and versatile file transfer utility commonly used to synchronize files and directories between different locations. It can transfer files over a network connection and run in various modes, including local and remote transfers and backup operations. One of the key benefits of using ``rsync`` is its ability to transfer only the differences between the source and destination files, which can significantly reduce the amount of data transfer time required. Additionally, ``rsync`` supports various advanced features, including the ability to perform incremental backups and preserve symbolic links, making it a popular tool for system administrators and other advanced users.
+
+.. important:: 
+   File transfers must be done using the transfer node on the Discovery, i.e., do not copy to or from the login node accessible via ``xfer.discovery.neu.edu``. See :ref:`transferring data` for more information.
```

Comment:
This statement could be reworded a bit or add in a comma because it seems like you get to the login node by ssh to xfer.